oat-sa/conventional-commit-action
# Feature/ADF-1417/Upgrade to node18

### Related to: [ADF-1417](https://oat-sa.atlassian.net/browse/ADF-1417) and [ADF-1419](https://oat-sa.atlassian.net/browse/ADF-1419)

Requires: 
 - [x] https://github.com/oat-sa/conventional-changelog-tao/pull/21
 - [x] update the file [`package.json`](https://github.com/oat-sa/conventional-commit-action/pull/18/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R23) once the companion PR has been merged and released
 - [x] delete the tag `v1`
 - [x] replace the tag `v0`

### Summary

Update the dependencies for supporting Node 18.

It also removes vulnerabilities 🎉 
```
audited 285 packages in 1.72s

found 0 vulnerabilities
```

### Details

To properly support Node 18, the dependencies had to be updated and aligned.

Version bumped from `0.2.0` to `0.4.0` because intermediate versions were added without a bump.

The files in the `dist` folder were generated from `npm run build`.

**Note:** The lock file was generated with `npm@6` so it would also work seamlessly with Node 14.

Addressed breaking change:
- https://github.com/actions/toolkit/blob/main/packages/github/RELEASES.md#500
    - `octokit.{endpoint}` must be replaced by `octokit.rest.{endpoint}`

### How to test
- make sure to have Node 18 installed (you can use [nvm.sh](https://github.com/nvm-sh/nvm) for managing multiple versions)
- install the package: `npm i`
- run the unit tests: `npm run build`
- test the linter: `npm run lint`

The tag `v1` has been temporarily created for checking the result. It has been set to [`oat-sa/tao-deliver-fe/.github/workflows/continuous-integration.yaml`](https://github.com/oat-sa/tao-deliver-fe/pull/745/files#diff-208542bbdcc8946dccfcfd7b7ac20705e8c5f382cdb3bd7fe6ecea4e3b9cf04bR20). You can check the report of the CI action in https://github.com/oat-sa/tao-deliver-fe/pull/745 
(quick ref: https://github.com/oat-sa/tao-deliver-fe/actions/runs/5001927825/jobs/8962173710#step:4:1)

[ADF-1417]: https://oat-sa.atlassian.net/browse/ADF-1417?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[ADF-1419]: https://oat-sa.atlassian.net/browse/ADF-1419?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
